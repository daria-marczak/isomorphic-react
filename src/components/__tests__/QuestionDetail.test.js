import { mapStateToProps, QuestionDetailDisplay } from "../QuestionDetail";
import renderer from "react-test-renderer";
import React from "react";

describe("The question detail component", () => {
  describe("container element", () => {
    describe("mapstateToProps", () => {
      it("should map the state to props correctly", () => {

        const sampleQuestion = {
          question_id: 42,
          body: "Lorem ipsum"
        };
        const appState = {
          questions: [sampleQuestion]
        };
        const ownProps = {
          question_id: 42
        };
        const componentState = mapStateToProps(appState, ownProps);
        expect(componentState).toEqual(sampleQuestion);
      });
    });
  });

  describe("The display element", () => {
    it("should not regress", () => {
      const tree = renderer.create(
        <QuestionDetailDisplay
          title="Something"
          body="Whatever"
          answer_count={5}
          tags={["blah blah"]}
        />
      );

      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});