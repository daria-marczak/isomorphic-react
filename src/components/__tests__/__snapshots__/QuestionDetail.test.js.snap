// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The question detail component The display element should not regress 1`] = `
<div>
  <h3
    className="mb-2"
  >
    Something
  </h3>
  <div>
    <div
      className="mb-3"
    >
      <div>
        <code>
          blah blah
        </code>
      </div>
    </div>
    <div>
      <p>
        Whatever
      </p>
    </div>
    <div>
      5
       Answers
    </div>
  </div>
</div>
`;
